plugins/nimble/personal_access_token.go

```diff
@@ -0,0 +1,52 @@
+package nimble
+
+import (
+	"github.com/1Password/shell-plugins/sdk"
+	"github.com/1Password/shell-plugins/sdk/importer"
+	"github.com/1Password/shell-plugins/sdk/provision"
+	"github.com/1Password/shell-plugins/sdk/schema"
+	"github.com/1Password/shell-plugins/sdk/schema/credname"
+	"github.com/1Password/shell-plugins/sdk/schema/fieldname"
+)
+
+func PersonalAccessToken() schema.CredentialType {
+	return schema.CredentialType{
+		Name:          credname.PersonalAccessToken,
+		DocsURL:       sdk.URL("https://github.com/nim-lang/nimble"),
+		ManagementURL: sdk.URL("https://nimble.directory/about.html"),
+		Fields: []schema.CredentialField{
+			{
+				Name:                fieldname.Token,
+				MarkdownDescription: "Token used to authenticate to GitHub.",
+				Secret:              true,
+				Composition: &schema.ValueComposition{
+					Length: 40,
+					Prefix: "ghp_",
+					Charset: schema.Charset{
+						Uppercase: true,
+						Lowercase: true,
+						Digits:    true,
+					},
+				},
+			},
+		},
+		DefaultProvisioner: provision.EnvVars(defaultEnvVarMapping),
+		Importer: importer.TryAll(
+			TryNimbleConfigFile(),
+			importer.TryEnvVarPair(defaultEnvVarMapping),
+		),
+	}
+}
+
+var defaultEnvVarMapping = map[string]sdk.FieldName{
+	"NIMBLE_GITHUB_API_TOKEN": fieldname.Token,
+}
+
+func TryNimbleConfigFile() sdk.Importer {
+	// TODO: Try importing from ~/.nimble/github_api_token file
```

Comment:
is this TODO a change to be carried on in this PR? If not, I would suggest adding an importer for a potentially already existing `NIMBLE_GITHUB_API_TOKEN` as a quick win, just so not to ship this with absolutely no importers.

Reply:
`importer.TryEnvVarPair(defaultEnvVarMapping),`

Reply:
Thanks so much for the feedback & suggestions, @AndyTitu !
I'll add in the env var importer this evening.

For the file importer, this is similar to [the HashiCorp Vault plugin](https://github.com/1Password/shell-plugins/blob/main/plugins/vault/auth_token.go#L49). I wasn't able to create a working version as the `~/.nimble/github_api_token` file doesn't have any formatting/extension type, just the token in plaintext.
I was hoping to ask for suggestions - I read through the other plugins to compare, but as far as I can see, [the package](https://pkg.go.dev/github.com/1Password/shell-plugins/sdk#ImportCandidate) seems to expect a key/value.

Reply:
I've pushed `TryEnvVarPair` to the top of the `TryAll` in importers. Cheers!

Reply:
Something like this should do the trick for the file importer, provided the file only contains the token:
```go
func TryNimbleConfigFile(path string) sdk.Importer {
	return importer.TryFile(path, func(ctx context.Context, contents importer.FileContents, in sdk.ImportInput, out *sdk.ImportAttempt) {
		credential := contents.ToString()
		fields := make(map[sdk.FieldName]string)
		if len(credential) != 0 {
			fields[fieldname.Token] = credential
		}
		out.AddCandidate(sdk.ImportCandidate{
			Fields: fields,
		})
	})
}
```

I didn't test this code, let us know in case this doesn't work and we can look deeper into it 😄

Reply:
Hey @hculea, thank you so much for your help on this! That seems to be working well & I've just tested that it can now import from that config file